cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+              new ResourcePattern(
+                  ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+
+          resultStr =
+              processAclBindings(
+                  clusterAclRequest,
+                  client,
+                  patternType,
+                  host,
+                  principal,
+                  aclListArray,
+                  aclEntry,
+                  resourcePattern);
+
+        } else if (isValidParam) {
+          List<AclBindingFilter> aclListArray = new ArrayList<>();
+
+          AccessControlEntryFilter aclEntry =
+              new AccessControlEntryFilter(
+                  principal, host, AclOperation.READ, AclPermissionType.ALLOW);
+          ResourcePatternFilter resourcePattern =
+              new ResourcePatternFilter(
+                  ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+
+          resultStr =
+              processOtherAclBindings(
+                  clusterAclRequest,
+                  client,
+                  patternType,
+                  host,
+                  principal,
+                  aclListArray,
+                  aclEntry,
+                  resourcePattern);
+        }
+      }
+
+      if (aclIp != null && aclIp.trim().length() > 0) {
+        aclIp = aclIp.trim();
+        host = aclIp;
+        principal = "User:*";
+
+        if (RequestOperationType.CREATE.equals(clusterAclRequest.getRequestOperationType())) {
+          List<AclBinding> aclListArray = new ArrayList<>();
+
+          ResourcePattern resourcePattern =
+              new ResourcePattern(
+                  ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+          AccessControlEntry aclEntry =
+              new AccessControlEntry(principal, host, AclOperation.READ, AclPermissionType.ALLOW);
+          resultStr =
+              processAclBindings(
+                  clusterAclRequest,
+                  client,
+                  patternType,
+                  host,
+                  principal,
+                  aclListArray,
+                  aclEntry,
+                  resourcePattern);
+
+        } else {
+          List<AclBindingFilter> aclListArray = new ArrayList<>();
+
+          ResourcePatternFilter resourcePattern =
+              new ResourcePatternFilter(
+                  ResourceType.TOPIC, clusterAclRequest.getTopicName(), patternType);
+          AccessControlEntryFilter aclEntry =
+              new AccessControlEntryFilter(
+                  principal, host, AclOperation.READ, AclPermissionType.ALLOW);
+          resultStr =
+              processOtherAclBindings(
+                  clusterAclRequest,
+                  client,
+                  patternType,
+                  host,
+                  principal,
+                  aclListArray,
+                  aclEntry,
+                  resourcePattern);
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
```suggestion
      log.error("Exception:", e);
```
it's better to log it into log rather than system output